### ISSUE TITLE
`overridden_fields` to consider `@mustBeOverridden`

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#3793]
**Describe the issue**
When a field is marked as `@mustBeOverridden`, [overridden_fields](https://dart-lang.github.io/linter/lints/overridden_fields.html) should ignore the fields in the sub-classes.

**To Reproduce**
```dart
import 'package:meta/meta.dart';

class A {
  @mustBeOverridden
  int f = 0;
}

class B extends A {
  @override
  int f = 0;
}
```

**Expected behavior**
No lint is triggered.

**Comment by lrhn** [comment]
Not necessarily true. You can, and likely should, override a field with a getter, not with another field. That's exactly what the lint is there to ensure.

If the problem is that the superclass has a concrete field, instead of an abstract field, that's an issue to file against them, but then you might want to use `implements` instead of `extends`.

**Comment by srawlins** [comment]
> You can, and likely should, override a field with a getter, not with another field. That's exactly what the lint is there to ensure.

Which lint rule? `overridden_fields`?

The documentation, while brief, just says, "it is bad practice to [override fields]." There are no GOOD examples of overriding a field with a getter or setter. 😮‍💨 😮‍💨 😮‍💨

https://dart-lang.github.io/linter/lints/overridden_fields.html

The implementation appears to support your statement, though. It only [checks fields](https://github.com/dart-lang/linter/blob/main/lib/src/rules/overridden_fields.dart#L130). And there are [no test cases](https://github.com/dart-lang/linter/blob/main/test_data/rules/overridden_fields.dart) of getters or setters which override fields. 😦

The lint diagnostic message is "Field overrides a field inherited from '{0}'." which supports your statement. The correction message, "Try removing the field, overriding the getter and setter if necessary." doesn't help. "But I did override the getter and setter! With a field!"
